web/app/js/components/util/TapUtils.jsx

```diff
@@ -0,0 +1,101 @@
+import _ from 'lodash';
+import { Popover } from 'antd';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+export const httpMethods = ["GET", "HEAD", "POST", "PUT", "DELETE", "CONNECT", "OPTIONS", "TRACE", "PATCH"];
+
+export const defaultMaxRps = "1.0";
+
+export const tapQueryPropType = PropTypes.shape({
+  resource: PropTypes.string,
+  namespace: PropTypes.string,
+  toResource: PropTypes.string,
+  toNamespace: PropTypes.string,
+  method: PropTypes.string,
+  path: PropTypes.string,
+  scheme: PropTypes.string,
+  authority: PropTypes.string,
+  maxRps: PropTypes.string
+});
+
+export const processTapEvent = jsonString => {
+  let d = JSON.parse(jsonString);
+  d.source.str = publicAddressToString(_.get(d, "source.ip.ipv4"), d.source.port);
+  d.destination.str = publicAddressToString(_.get(d, "destination.ip.ipv4"), d.destination.port);
+
+  d.source.pod = _.has(d, "sourceMeta.labels.pod") ? "po/" + d.sourceMeta.labels.pod : null;
+  d.destination.pod = _.has(d, "destinationMeta.labels.pod") ? "po/" + d.destinationMeta.labels.pod : null;
+
+  switch (d.proxyDirection) {
+    case "INBOUND":
+      d.tls = _.get(d, "sourceMeta.labels.tls", "");
+      break;
+    case "OUTBOUND":
+      d.tls = _.get(d, "destinationMeta.labels.tls", "");
+      break;
+    default:
+      // too old for TLS
+  }
+
+  if (_.isNil(d.http)) {
+    this.setState({ error: "Undefined request type"});
+  } else {
+    if (!_.isNil(d.http.requestInit)) {
+      d.eventType = "requestInit";
```

Comment:
setting the event type to `requestInit` rather than `req` does mean some of our later accessors look like `requestInit.http.requestInit.path` instead of `req.http.requestInit.path`, but I think it's still clearer, and avoids a little repetition here.